File: app/src/main/java/com/drkj/foxconn/BaseActivity.java
package com.drkj.foxconn;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import com.drkj.foxconn.util.NfcCardUtil;
import com.zltd.decoder.DecoderManager;

/**
 * Created by ganlong on 2018/1/16.
 * Modify by VeronicaRen on 2018/3/14.
 */
public abstract class BaseActivity extends AppCompatActivity {

    protected NfcCardUtil nfcCardUtil;

    protected boolean isResume = false;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        nfcCardUtil = new NfcCardUtil(this);
    }

    @Override
    protected void onStart() {
        super.onStart();
    }

    @Override
    protected void onResume() {
        super.onResume();
        isResume = true;
        nfcCardUtil.startNfc();
    }

    @Override
    protected void onPause() {
        super.onPause();
        isResume = false;
        nfcCardUtil.stopNfc();
    }
}

File: app/src/main/java/com/drkj/foxconn/activties/BaseActivity.kt
package com.drkj.foxconn.activties

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.drkj.foxconn.util.SpUtil

/**
 * 基础Activity
 * Created by VeronicaRen on 2018/2/23 in Kotlin
 */
abstract class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(setLayout())
        initView()
    }

    abstract fun setLayout(): Int

    abstract fun initView()

    fun getToken(): String = SpUtil.getString(this, "token")
}